### PR TITLE
tests: Ensure that parseOptionalTime honors nanoseconds

### PR DESCRIPTION
**Pull request by sergio-costas**
This PR adds a test for parseOptionalTime, which checks whether it honors nanoseconds in a RFC3339 date/time or not.

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by ZeyadYasser** [comment]
Added `Skip spread` label since it is only touching unit test code

File: daemon/request_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+	"github.com/snapcore/snapd/daemon"
+	"gopkg.in/check.v1"
+)
+
+type requestSuite struct{}
+
+var _ = check.Suite(&requestSuite{})
+
+func (s *requestSuite) TestParseDateHasNanosecondPrecision(c *check.C) {
```
[the remark this answers is not in the document]
Comment (sergio-costas) [reply daemon/request_test.go]: Good point.

File: daemon/export_request_test.go
[the remark this answers is not in the document]
Comment (sergio-costas) [reply daemon/export_request_test.go:1]: Good point!

**Review by Meulengracht — APPROVED** [review]
Thank you

**Pull request merged**